manager/src/test/java/com/redhat/service/smartevents/manager/ProcessorServiceConnectorTest.java

```diff
@@ -296,7 +296,9 @@ private void assertShardAsksForProcessorToBeDeletedIncludes(Processor processor)
 
     private void assertShardAsksForProcessorToBeDeletedDoesNotInclude(Processor processor) {
         List<Processor> processorsToBeDeleted = processorDAO.findByShardIdToDeployOrDelete(TestConstants.SHARD_ID);
-        assertThat(processorsToBeDeleted.stream().map(Processor::getId)).doesNotContain(processor.getId());
+        boolean empty=processorsToBeDeleted.isEmpty();
+        if(!empty){
+        assertThat(processorsToBeDeleted.stream().map(Processor::getId)).doesNotContain(processor.getId());}
```

Comment:
Does it mean that the value can be empty list or non empty list, based on which test called this method?
In such case it may have sense either to separate those cases into separate methods, or use the code directly in the caller test if only one test is using this check.

A general rule - the tests ideally shouldn't contain `if`, they should be deterministic.

Reply:
Ohkay, will try to modify and will remove the usage of `if`. Thanks.

> Does it mean that the value can be empty list or non empty list, based on which test called this method? In such case it may have sense either to separate those cases into separate methods, or use the code directly in the caller test if only one test is using this check.
> 
> A general rule - the tests ideally shouldn't contain `if`, they should be deterministic.